compiler/cpp/src/generate/t_java_generator.cc

```diff
@@ -2579,7 +2580,11 @@ void t_java_generator::generate_service_future_interface(t_service* tservice) {
   vector<t_function*> functions = tservice->get_functions();
   vector<t_function*>::iterator f_iter;
   for (f_iter = functions.begin(); f_iter != functions.end(); ++f_iter) {
-    indent(f_service_) << function_signature_future(*f_iter) << ";" << endl << endl;
+    t_function* t_f = *f_iter;
+    int mask;
+    for (mask = 0; mask < (1 << t_f->get_arglist()->get_members().size()); mask++) {
```

Comment:
Could we make this an option? If we want to push this upstream, I imagine it makes sense to let people choose between "only one function that only takes futures" and "the 2^n varations".